### DISCUSSION TITLE
Question about templates: how to search for an instance at an unknown address?

### DISCUSSION THREAD

**Discussion by markaudacity** [HexFiend/HexFiend#397] [Q&A]
I'm trying to write templates for some old music hardware to parse their backup files, and I find myself needing to search for data that does not always occur at the same location in the file. The data is in a known format, but occurs an arbitrary number of times at arbitrary locations, so I can't simply move the pointer to a location.

How do I do this?

**Comment by kainjow** [comment]
There is not a built-in mechanism for this. You'd have to do this with a combination of moving the file pointer, maybe painstakingly byte-by-byte, and reading in whatever format you'd expect with some validation. Are you sure it's arbitrary, or maybe the format isn't fully known?
